### PR TITLE
Add docs and tidy up

### PR DESCRIPTION
**Pull request by rlh1994**
- [x] readme
- [x] dbt docs
- [x] macro docs
- [x] requirements.txt
- [x] tidy project yaml
- [ ] other typos and small fixes

Note that all links in docs point to the snowplow org on github, not the incubator to avoid changing them all when this is moved.

### CODE REVIEW

File: docs/markdown/snowplow_event_splitting_overview.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+*Note this model design doc site is linked to latest release of the package. If you are not using the latest release, [generate and serve](https://docs.getdbt.com/reference/commands/cmd-docs#dbt-docs-serve) the doc site locally for accurate documentation.*
+
+## Overview
+
+This package consists of two macros, a python script, and some example configuration files to help you get started:
+
+  - `split_events` _(macro)_: This macro does the heavy lifting of the package, taking a series of inputs to generate the SQL required to split the events table and flatten (1 level) of any [self-describing event][self-desc-events] or [context][sp-contexts] columns. While you can use this macro manually it is recommended to create the models that use it by using the script provided.
+
+  - `users_table` _(macro)_: This macro takes a series of inputs to generate the SQL that will produce your users table (1 row per user, with the latest values of the contexts you specify as the other columns), using the `user_id` column and any custom contexts from your events table.
```
Comment (emielver) [inline docs/markdown/snowplow_event_splitting_overview.md:19]: Might be worth mentioning here that latest values is based on `collector_tstamp`

**Review by emielver — COMMENTED** [review]
Looks great, really shaping up well!

File: README.md [hunk 2]
```diff
@@ -1,15 +1,82 @@
...
+- Provides macros to simplify the production of models that split Snowplow event data into a table per event type, plus a reduced events table and latest user context table, for easier integration with downstream tools.
+- Includes a python script to generate the models using a simple configuration file that uses the schemas used within your pipelines to identify the columns, to greatly reduce the upfront effort in creating these models.
+
+Please refer to the [doc site](https://docs.snowplow.io/docs/modeling-your-data/modeling-your-data-with-dbt/) for a full breakdown of the package.
+
+> Note that currently only data processed by the [Snowplow Web dbt package](https://github.com/snowplow/dbt-snowplow-web) is supported and any filters applied within this package will be also applied here e.g. `app_id`.
+
+### Adapter Support
+
+The snowplow-event-splitter v0.1.0 package currently supports BigQuery, Databricks & Snowflake.
```
[the remark this answers is not in the document]
Comment (rlh1994) [reply README.md:18]: Let's chat about this next week

File: README.md [hunk 3]
```diff
@@ -1,15 +1,82 @@
...
+
+Check dbt Hub for the latest installation instructions, or read the [dbt docs][dbt-package-docs] for more information on installing packages.
+
+### Configuration & Operation
+
+Please refer to the [doc site][snowplow-event-splitter-docs] for details on how to configure and run the package.
+
+### Models
+
+This package doesn't contain any models directly, however it is used to produce 3 types of models:
```
[the remark this answers is not in the document]
Comment (rlh1994) [reply README.md:41]: I'm going to add some more to the full docs site to make it clearer so will re-point them to there

File: docs/markdown/snowplow_event_splitting_overview.md (hunk 1 above)
Comment (rlh1994) [reply docs/markdown/snowplow_event_splitting_overview.md:19]: Agree, going to match this with the new bits to the docs site, but going to do it a new branch along with the repo name change to keep it all a bit cleaner.

**Pull request merged**